plugins/storage/volume/linstor/src/main/java/org/apache/cloudstack/storage/datastore/driver/LinstorPrimaryDataStoreDriverImpl.java

```diff
@@ -958,9 +954,12 @@ private Answer copyTemplate(DataObject srcData, DataObject dstData) {
             }
             else {
                 answer = new Answer(cmd, false, "Unable to get matching Linstor endpoint.");
+                deleteResourceDefinition(pool, rscName);
             }
         } catch (ApiException exc) {
             s_logger.error("copy template failed: ", exc);
+            s_logger.info(String.format("deleting failed template resource: %s", rscName));
```

Comment:
can move this log to `deleteResourceDefinition()` method?

Reply:
done